org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/Document.xtend

```diff
@@ -119,31 +119,36 @@ import org.eclipse.lsp4j.Range
     }
 
     def String getSubstring(Range range) {
-    		val start = getOffSet(range.start)
-    		val end = getOffSet(range.end)
-    		return this.contents.substring(start, end)
+          val start = getOffSet(range.start)
+          val end = getOffSet(range.end)
+          return this.contents.substring(start, end)
     }
 
     def Document applyChanges(Iterable<? extends TextEdit> changes) {
```

Comment:
Could you please add documentation that explicitly states the relation of the offsets / ranges in the iterable of TextEdits, e.g. that the second text edit has a range that relates to the result after applying the first edit?

Reply:
Done